pkg/web/api/connector_v1.go

```diff
@@ -65,6 +67,41 @@ func (c *ConnectorAPIv1) ListConnectors(
 	return &apiv1.ListConnectorsResponse{Connectors: clist}, nil
 }
 
+func (c *ConnectorAPIv1) InspectConnector(req *apiv1.InspectConnectorRequest, server apiv1.ConnectorService_InspectConnectorServer) error {
```

Comment:
I have a problem with having this much logic in the gRPC handler, these should normally only care about parsing requests and formatting responses. I'd rather put this logic in the orchestrator or service, something like:
```
// call orchestrator to start inspecting a connector, get back a channel of records
// that will be closed when the context is closed or the pipeline stops
// if the pipeline is not running an error is returned instead
recordChan, err := c.cs.Inspect(server.Context(), req.Id)
if err != nil { ... }
for r := range recordChan { ... }
```

Reply:
That's a nice suggestion, thanks. :+1: